app/aws/s3.py

```diff
@@ -60,8 +60,13 @@ def get_job_metadata_from_s3(service_id, job_id):
     return obj.get()["Metadata"]
 
 
-def remove_job_from_s3(service_id, job_id):
-    return remove_s3_object(*get_job_location(service_id, job_id))
+# AWS has a limit of 1000 objects per delete_objects call
+def remove_jobs_from_s3(jobs, batch_size=1000):
+    bucket = resource("s3").Bucket(current_app.config["CSV_UPLOAD_BUCKET_NAME"])
+
+    for start in range(0, len(jobs), batch_size):
+        object_keys = [FILE_LOCATION_STRUCTURE.format(job.service_id, job.id) for job in jobs[start : start + batch_size]]
+        bucket.delete_objects(Delete={"Objects": [{"Key": key} for key in object_keys]})
```

Comment:
Added a loop here so we won't go over the AWS limit in the future. Added a test for this function below.